app/models/nextUpdates/NextUpdatesResponseModel.scala

```diff
@@ -53,6 +54,26 @@ case class ObligationsModel(obligations: Seq[NextUpdatesModel]) extends NextUpda
 
   def obligationsByDate(implicit mtdItUser: MtdItUser[_]): Seq[(LocalDate, Seq[NextUpdateModelWithIncomeType])] =
     allDeadlinesWithSource().groupBy(_.obligation.due).toList.sortWith((x, y) => x._1.isBefore(y._1))
+
+  def groupByQuarterPeriod(obligations: Seq[NextUpdateModelWithIncomeType]): Map[String, Seq[NextUpdateModelWithIncomeType]] = {
```

Comment:
* there is 3 lines to do with the sorting in this method;
* :seedling:: might be worth to remove sorting at all from any business logic like this method ? as this is something to do with the views (presentation/rendering layer);
* :wrench: lines 60-65 can me moved into separate function ?